content/markdown/02-working-with-data/02-05-new-columns.rmd

```diff
@@ -0,0 +1,151 @@
+---
+output:
+  pdf_document: default
+  html_document: default
+---
+# Making New Columns
+**Author:** Cole Brookson
+**Date:** 20 July 2022
+
+```{r, echo=FALSE}
+options(tidyverse.quiet = TRUE)
+```
+
+While summarizing columns is useful if we want to pare down our dataframe and get a summary, it's often the case we actually just want to add a new column to our dataframe, perhaps based on some set of other columns already in our dataframe. 
+
+In Tidy R, we will almost always use `dplyr::mutate()` for this task. 
+
+## Mutating Columns with Single Values
+
+The simplest use of `mutate()` is to make a new column that is made up of either a pre-existing object or a single value. For examples here, let's turn to a long-term dataset - sugar maple seedlings at Hubbard Brook Experimental Forest. 
+
+```{r}
+library(tidyverse)
+library(lterdatasampler)
+df = lterdatasampler::hbr_maples
+head(df)
+```
+For this example, we only will work with some of the variables, so let's first select the columns we want (LINK TO SECTION ON SELECTION). 
+
+```{r}
+df = df %>% 
+  dplyr::select(year, watershed, elevation, leaf1area, 
+                leaf2area, corrected_leaf_area)
+df
+```
+
+
+We could for example, want a column in this dataset that tells us what forest these data are from. That would be an easy use of the `mutate()` function as there's only one value. We could do this like so:
+
+```{r}
+df = df %>% 
+  dplyr::mutate(forest = "HBEF")
+df
+```
+
+And we see that our new column has been added onto the far right of the dataframe. 
+
+## Mutating Columns as a Function of Existing Columns
+
+A more common use case is if we want to make a new column from existing columns. Two typical uses here would be:
+
+1. Making a new column that is made via a function operating on another column(s)
+
+2. Making a factor column that binds together values from other columns 
+
+### 1 - New Column Via Numeric Function or Logical Test
+
+#### Rowwise Function
+
+In these data, there are two measures of leaf area: `leaf1area` and `leaf2area`. While they're both very similar, it might still be useful to have a mean estimate for each observation in the data between the two leaf area measurements. 
+
+Say we wanted to calculate for row, the mean value of the two measures of leaf area. **To make this operation on a row-wise basis, we can use `dplyr::rowwise()`** which will ensure each calculation is done on a single row. Combining this with `mutate()` will produce our desired result. 
```

Comment:
I might edit the first sentence to read: "Say we wanted to calculate the mean value of the measures of leaf area for each row."